core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -1206,4 +1206,61 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
       return ExprEval.of(expr.value() != null, ExprType.LONG);
     }
   }
+
+  class LpadFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "lpad";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() != 3) {
+        throw new IAE("Function[%s] needs 3 arguments", name());
+      }
+      
+      String base = NullHandling.nullToEmptyIfNeeded(args.get(0).eval(bindings).asString());
+      int len = args.get(1).eval(bindings).asInt();
+      String pad = NullHandling.nullToEmptyIfNeeded(args.get(2).eval(bindings).asString());
```

Comment:
Does this make sense for the `pad`? Seems like it should maybe be an error if it's null?

Reply:
@clintropolis Thanks for review. When base or pad is null, it does not make sense to do these.